docs/sdk/src/reference_docs/frame_origin.rs

```diff
@@ -1,14 +1,263 @@
+//! [`frame::runtime::prelude::EnsureNone`], etc. But, there are also many more that are not known
+//! to us, and are defined in other pallets.
+//!
+//! For example, [`pallet_collective`] defines [`pallet_collective::EnsureMember`] and
+//! [`pallet_collective::EnsureProportionMoreThan`] and many more, which is exactly what we alluded
+//! to earlier in this document.
+//!
+//! Make sure to check the full list of [implementors of
+//! `EnsureOrigin`](frame::traits::EnsureOrigin#implementors) for more inspiration.
+//!
+//! ## Obtaining Abstract Origins
+//!
+//! So far we have learned that FRAME pallets can assert on custom and abstract origin types,
+//! whether they are defined within the pallet or not. But how can we obtain these abstract origins?
+//!
+//! > All extrinsics that come from the outer world can generally only be obtained as either
+//! > `signed` or `none` origin
+//!
+//! Generally, these abstract origins are only obtained within the runtime, when a call is
+//! dispatched within the runtime.
+//!
+//! ## Further References
+//!
+//! - [Gavin Wood's speech about FRAME features at Protocol Berg 2023.](https://youtu.be/j7b8Upipmeg?si=83_XUgYuJxMwWX4g&t=195)
+//! - [A related StackExchange question.](https://substrate.stackexchange.com/questions/10992/how-do-you-find-the-public-key-for-the-medium-spender-track-origin)
+//!
+//! [^1]: Inherents are essentially unsigned extrinsics that need an [`frame_system::ensure_none`]
+//! origin check, and through the virtue of being an inherent, are agreed upon by all validators.
+
+use frame::prelude::*;
+
+#[frame::pallet(dev_mode)]
+pub mod pallet_for_origin {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[docify::export(call_simple)]
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		pub fn do_something(_origin: OriginFor<T>) -> DispatchResult {
+			//              ^^^^^^^^^^^^^^^^^^^^^
+			todo!();
+		}
+	}
+}
+
+#[frame::pallet(dev_mode)]
+pub mod pallet_with_custom_origin {
+	use super::*;
+
+	#[docify::export(custom_origin_bound)]
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeOrigin: From<<Self as frame_system::Config>::RuntimeOrigin>
+			+ Into<Result<Origin, <Self as Config>::RuntimeOrigin>>;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[docify::export(custom_origin)]
+	/// A dummy custom origin.
+	#[pallet::origin]
+	#[derive(PartialEq, Eq, Clone, RuntimeDebug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+	pub enum Origin {
+		/// If all holders of a particular NFT have agreed upon this.
+		AllNftHolders,
+		/// If all validators have agreed upon this.
+		ValidatorSet,
+	}
+
+	#[docify::export(custom_origin_usage)]
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		pub fn only_validators(
+			// we use the full syntax of `OriginFor` here, but it's not necessary.
```

Comment:
What does "full syntax" mean here? Leftover comment and was previously using full crate path?

Reply:
yeah old comment.